### ISSUE TITLE
Simulating a "blackbox" FMU causes simulation to crash

### ISSUE THREAD

**Issue by dietmarw** [OpenModelica/OpenModelica#8158]
### Description

When testing out the FMU handling of OMEdit I noticed that one can not simulate an FMU (exported by OMEdit itself) when it is of type "blackbox". Doing so results in 

```
Process crashed Simulation process failed. Exited with code 6.
/tmp/SimFMU/SimFMU -port=36155 -logFormat=xmltcp -override=startTime=0,stopTime=1,stepSize=0.002,tolerance=1e-6,solver=dassl,outputFormat=mat,variableFilter=.* -r=/tmp/SimFMU/SimFMU_res.mat -w -lv=LOG_STATS -inputPath=/tmp/SimFMU -outputPath=/tmp/SimFMU
Process crashed
getBestJumpBuffer got mmc_jumper=(nil), globalJumpBuffer=(nil)
```
### Steps to Reproduce
I*ve created the blackbox FMU `TestFMU.fmu` from the model `TestFMU.mo` and run both side by side in `SimFMU.mo`. All files are included in: 
[TestFMUs.zip](https://github.com/OpenModelica/OpenModelica/files/7554284/TestFMUs.zip)

### Expected Behavior

Simulation should work just like it does if one chooses one of the other non-blackbox type FMUs when exporting.

### Screenshots
Test FMU's content: 
![image](https://user-images.githubusercontent.com/9332/142193098-9f7638da-e2a3-4c6e-b3bd-f22c6d7f2921.png)

### Version and OS
64bit Ubuntu Linux 20.04
OMEdit - OpenModelica Connection Editor
Connected to OpenModelica 1.19.0~dev-375-g200aa9b

**Comment by casella** [comment]
@lochel, can you please have a look?

**Comment by dietmarw** [comment]
I've also tested the FMU with FMPy and they also expect to find `TestFMU_info.json` also in a blackbox FMU. So could the problem be on the generation side rather than on the importing side?

**Comment by dietmarw** [comment]
I ran some more tests. Exporting the same model with help of Dymola also does not contain `TestFMU_info.json ` but can be simulated without problems in FMPy. Not so when importing in OMEdit. (could not test in OMSimulator because of #8163 not being resolved yet). 

So it looks like the problem is probably not the missing `TestFMU_info.json` itself but something inside the FMU that maye tries to point to that file. 

I also imported the OMC generated FMU in Dymola which then fails with:
```
Error: The following error was detected at time: 0
/home/dietmarw/.workspace/Resources/Library/FMU/FMUtests.SinSource/binaries/linux64/TestFMU_FMU.libs: file too short
The stack of functions is:
FMUtests_SinSource_fmu.fmi_Functions.fmiModel
FMUtests_SinSource_fmu.fmi_Functions.fmiModel("FMUtests_SinSource_fmu", fMUtests_SinSource_fmu.fmi_loggingOn, "file:////home/dietmarw/.workspace/Resources/Library/FMU/FMUtests.SinSource/resources", 0, 1)
Error: Integrator failed to start model.
... Error message from dymosim
ERROR: The simulation of Unnamed FAILED
```
Maybe that helps identifying the cause of the corrupt FMU generation.

**Comment by dietmarw** [comment]
I just noticed that my continued tests were now done with a slightly different TestFMU:

![image](https://user-images.githubusercontent.com/9332/142618092-d425942b-01e0-4193-a902-432d30a901e3.png)

**Comment by dietmarw** [comment]
The question is also is that json needed at all. Like I showed in #8411, Dymola does not even contain that file for both blackbox and protected.

**Comment by arun3688** [comment]
@dietmarw The issue is fixed with this commit https://github.com/OpenModelica/OpenModelica/commit/808cd71bd1357ceba4d54cfc12ea78754eae1565, You can test the fixes from the nightly builds tomorrow

**Comment by dietmarw** [comment]
I can confirm that this issue has now been fixed by #8433

**Comment by casella** [comment]
Thanks @dietmarw for reporting and to @arun3688 for fixing.
